### PR TITLE
Beamline windows

### PR DESCRIPTION
**Pull request by willrogers**
@nickbattam not for merging

### CODE REVIEW

File: plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/OpenNewWindow.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import org.csstudio.openfile.IOpenDisplayAction;
+
+public class OpenNewWindow implements IOpenDisplayAction {
+
+    @Override
+    public void openDisplay(String path, String data) {
+        Plugin.getLogger().config("Opening " + path + " with data " + data);
+        Path p = Paths.get(path);
+        if (Files.exists(p)) {
```
Comment (nickbattam) [inline plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/OpenNewWindow.java]: No warning logged if path doesn't exist

File: plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowOpener.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        if (!found) {
+            try {
+                PlatformUI.getWorkbench().openWorkbenchWindow(windowSpec.getPerspectiveId(), null);
+            } catch (WorkbenchException e) {
+                Plugin.getLogger().log(Level.WARNING, "Failed to open new workbench window", e);
+            }
+        }
+    }
+
+    public void forceActive(Shell shell) {
```
Comment (nickbattam) [inline plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowOpener.java]: lots of commented out code here

File: plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowSpecParser.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public WindowSpec get() {
+        return windowSpec;
+    }
+
+    private String getTextValue(Element doc, String tag) {
+        String value = null;
+        NodeList nl;
+        nl = doc.getElementsByTagName(tag);
```
Comment (nickbattam) [inline plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowSpecParser.java]: why on two lines?

File: plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowSpecParser.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        windowSpec = spec;
+    }
+
+    public WindowSpec get() {
+        return windowSpec;
+    }
+
+    private String getTextValue(Element doc, String tag) {
+        String value = null;
+        NodeList nl;
```
Comment (nickbattam) [inline plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowSpecParser.java]: 'nodes' would be a easier name to read than n1

File: plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowSpecParser.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            String pid = getTextValue(doc, perspectiveId);
+            if (pid == null) {
+                throw new WindowManagementException("PerspectiveId is required.");
+            }
+            spec.setPerspectiveId(pid);
+            spec.setPerspectiveFile(getTextValue(doc, perspectiveFile));
+            NodeList linksElements = dom.getElementsByTagName(linksTag);
+            Node linksElement = linksElements.item(0);
+            if (linksElement != null) {
+                spec.setPerspectiveId(pid);
```
Comment (nickbattam) [inline plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowSpecParser.java]: this has already been set

File: plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowSpecParser.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                NodeList linkList = linksElement.getChildNodes();
+                System.out.println("There are " + linkList.getLength() + " element in the linklist");
+                for (int i = 0; i < linkList.getLength(); i++) {
```
Comment (nickbattam) [inline plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowSpecParser.java]: do you need this index?

File: plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/PerspectiveHelper.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        }
+    }
+
+    public void loadPerspective(String perspectiveFile) throws WindowManagementException {
+        FileUtils fu = new FileUtils();
+        IEclipseContext context = PlatformUI.getWorkbench().getService(IEclipseContext.class);
+        context.set(IFileUtils.class.getCanonicalName(), fu);
+        PerspectiveLoader loader = ContextInjectionFactory.make(PerspectiveLoader.class, context);
+        Path perspectiveFilePath = Paths.get(perspectiveFile);
+        URI fileUri = fu.pathToEmfUri(Paths.get(perspectiveFile));
```
Comment (nickbattam) [inline plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/PerspectiveHelper.java]: reuse perspectiveFilePath

File: plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowSpec.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    public void setPerspectiveFile(String perspectiveFile) {
+        this.perspectiveFile = perspectiveFile;
+    }
+
+    public void addLink(String eclipsePath, String filePath) {
+        links.put(eclipsePath, filePath);
+    }
+
+    public Map<String, String> getLinks() {
+        return links;
```
Comment (nickbattam) [inline plugins/org.csstudio.openfile.newwindow/src/org/csstudio/openfile/newwindow/WindowSpec.java]: return a copy of links

File: plugins/org.csstudio.openfile.newwindow.test/src/org/csstudio/openfile/beamlines/WindowSpecParserTest.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
+package org.csstudio.openfile.beamlines;
```
Comment (nickbattam) [inline plugins/org.csstudio.openfile.newwindow.test/src/org/csstudio/openfile/beamlines/WindowSpecParserTest.java]: wrong namespace?

**Comment by willrogers** [comment]
@nickbattam significant refactoring...

**Comment by nickbattam** [comment]
@willrogers

**Pull request merged**